src/main/java/net/dries007/tfc/util/events/StartFireEvent.java

```diff
@@ -49,6 +86,70 @@ public static boolean startFire(Level level, BlockPos pos, BlockState state, Dir
 
         if (!cancelled && event.isStrong())
         {
+            final BlockPos abovePos = pos.above();
+            // Check conditions for creating a firepit if a valid firestarter
+            if (FirepitBlock.canSurvive(level, abovePos) && firepitBaseChance != -1)
+            {
+                final List<ItemEntity> items = level.getEntitiesOfClass(ItemEntity.class, new AABB(abovePos.getX() - 0.5, abovePos.getY(), abovePos.getZ() - 0.5, abovePos.getX() + 1.5, abovePos.getY() + 1, abovePos.getZ() + 1.5));
+                final List<ItemEntity> usableItems = new ArrayList<>();
```

Comment:
Okay so a getEntities calculation is expensive and I would rather perform this later rather than sooner. It's cheaper to check the blockstate/entity than to get entities from the global list and filter them by location and class. Let's also extrac this function to be separate for cleanliness

Reply:
How does one check the blockstate/entity? This is all just code I moved from the firestarter item class, so it is what we were using before.

As far as extracting the function, that proved trickier than I thought as well because we either return true if the firepit is lit successfully, false if it could be lit but failed, or continue if 3 sticks and a log are not found. of course there are ways to get around that, but the ones that spring to mind are kind of ugly